Add license lookup for datasets

ethanwhite opened the pull request:
Implements a function and CLI for looking up dataset licenses from a
field containing a single license following the `licenses` property
in the frictionless data datapackage spec:
https://specs.frictionlessdata.io/data-package/#licenses

Also adds licenses for some datasets.

ethanwhite commented:
Implements #899.

I decided to submit this against `python-interface2` so that I couple implement it in both interface and CLI instead of putting that off.

I've tried to design it in such a way that we can easily move `citation` over to the same basic structure and location.

henrykironde commented on retriever/lib/datasets.py:
```diff
@@ -28,3 +35,12 @@ def datasets(arg_script=None):
 
     return all_scripts
+
+def license(dataset):
```
scripts = script_dict()
This could be a function that test if a script is available.
It will be used by both `ls` and `citation` and `license`.
The function could take a string and convert it to lowercase.
An in the function we could handle the errors.


```
    if dataset in scripts:
        script = scripts[dataset]
    else:
        raise UnknownDataset(dataset)
```

ethanwhite replied:
Agreed. It could also be used as a check in `install`, `download`, etc. Are you envisioning a structure like:

```
if dataset_exists(dataset):
    do stuff
```

or we could go with:

```
check_dataset_exists()
do stuff
```

And have the function throw the error?

ethanwhite commented:
Based on feedback from @henrykironde both here and on Slack I've refactored the PR as follows:

1. Dropped the special error class and now simply thrown and informative `KeyError`
2. Combined error handling and script lookup into a single `get_script` function that takes the name of a dataset as an argument, returns the associate script if found, and otherwise throws an error
3. Instead of returning `'Not currently available'` in the Python interface if a license doesn't exist, return `None`. Handle the information on available in the print statement for the CLI.

@ShivamNegi - I'd welcome a review from you on this as well both to get your suggestions for improvement and so that you can see how I'm thinking about handling printing (via the CLI) vs. returning programmatically usable information (in the Python interface).

henrykironde commented:
@ethanwhite, Sorry for taking long. I asked @ShivamNegi to make some changes that we could add to this PR and he is working on them. I Will give you feedback when we come up with the changes.
